### PR TITLE
IL-1383 Stanley fix

### PR DESCRIPTION
**Pull request by mario-ipor**

### CODE REVIEW

File: scripts/transfer_ownership_dai.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    await impersonatedAdminAddress.sendTransaction({
+        to: timelockAddress,
+        value: ethers.utils.parseEther("1.0"), // Sends exactly 1.0 ether
+    });
+
+    await helpers.impersonateAccount(timelockAddress);
+    const impersonatedSigner = await ethers.getSigner(timelockAddress);
+
+    let josephDaiContract = new hre.ethers.Contract(
```
Comment (pete-ipor) [inline scripts/transfer_ownership_dai.js:37]: const ?

**Review by pete-ipor — APPROVED** [review]

File: scripts/transfer_ownership_dai.js (hunk 1 above)
Comment (mario-ipor) [reply scripts/transfer_ownership_dai.js:37]: done

File: contracts/vault/Stanley.sol [hunk 2]
```diff
@@ -107,115 +114,79 @@ abstract contract Stanley is
...
-
-            return (withdrawnAmount, balance);
-        }
-
-        if (assetBalanceAave < assetBalanceCompound) {
-            uint256 ivTokenAmountPart = IporMath.division(
-                assetBalanceCompound * Constants.D18,
-                exchangeRate
+        if (selectedWithdrawAmount > 0) {
+            //Tranfer from Strategy to Stanley
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply contracts/vault/Stanley.sol:164]: done

File: hardhat.network.ts [hunk 3]
```diff
@@ -37,62 +35,12 @@ if (fork_enabled == "true") {
 }
 
 networks.hardhatfork = {
-    chainId: 1,
+    chainId: 31337,
     url: "http://127.0.0.1:8545",
     forking: {
-        url: `https://eth-mainnet.alchemyapi.io/v2/rT2R1mRGMzUr80dcTdwOTG1JSMVYzSJi`,
-        blockNumber: 14222088, // 14222087 Compaund APY: 2093206012009920000, AAVE APY: 2242773611830780298
+        url: `https://eth-mainnet.g.alchemy.com/v2/ufaEYojUhakRot1riVa0TIJ8OzfqiLhZ`,
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply hardhat.network.ts]: done

File: scripts/transfer_ownership_usdt.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    await impersonatedAdminAddress.sendTransaction({
+        to: timelockAddress,
+        value: ethers.utils.parseEther("1.0"), // Sends exactly 1.0 ether
+    });
+
+    await helpers.impersonateAccount(timelockAddress);
+    const impersonatedSigner = await ethers.getSigner(timelockAddress);
+
+    let josephUsdtAddress = "0x33C5A44fd6E76Fc2b50a9187CfeaC336A74324AC";
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply scripts/transfer_ownership_usdt.js:22]: done

File: scripts/transfer_ownership_usdc.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    await impersonatedAdminAddress.sendTransaction({
+        to: timelockAddress,
+        value: ethers.utils.parseEther("1.0"), // Sends exactly 1.0 ether
+    });
+
+    await helpers.impersonateAccount(timelockAddress);
+    const impersonatedSigner = await ethers.getSigner(timelockAddress);
+
+
+    let josephUsdcAddress = "0xC52569b5A349A7055E9192dBdd271F1Bd8133277";
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply scripts/transfer_ownership_usdc.js:23]: done

File: scripts/mint_usdc.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+const hre = require("hardhat");
+const ethers = hre.ethers;
+const usdcAbi = require("../abis/usdcAbi.json");
+async function main() {
+    const address = "0xFcb19e6a322b27c06842A71e8c725399f049AE3a";
+    await helpers.impersonateAccount(address);
+    const impersonatedSigner = await ethers.getSigner(address);
+    let usdcContract;
+    let usdcAddress = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48";
+    usdcContract = new hre.ethers.Contract(usdcAddress, usdcAbi, impersonatedSigner);
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply scripts/mint_usdc.js:11]: done

File: scripts/mint_usdc.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+const helpers = require("@nomicfoundation/hardhat-network-helpers");
+const hre = require("hardhat");
+const ethers = hre.ethers;
+const usdcAbi = require("../abis/usdcAbi.json");
+async function main() {
+    const address = "0xFcb19e6a322b27c06842A71e8c725399f049AE3a";
+    await helpers.impersonateAccount(address);
+    const impersonatedSigner = await ethers.getSigner(address);
+    let usdcContract;
+    let usdcAddress = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48";
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply scripts/mint_usdc.js:10]: done

File: app/src/drizzleOptions.js [hunk 8]
```diff
@@ -1,3 +1,5 @@
+import * as fs from "fs";
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply app/src/drizzleOptions.js:1]: no, done

File: scripts/validation.js [hunk 9]
```diff
@@ -6,7 +6,7 @@ async function main() {
     signer = await hre.ethers.provider.getSigner(await accounts[0].getAddress());
 
     const Migrations = await ethers.getContractFactory("Migrations", signer);
-    const migrations = Migrations.attach("0x93d0f54207a7A1bDAaE81D2e80320f3a7903750C");
+    const migrations = Migrations.attach("0x987e855776C03A4682639eEb14e65b3089EE6310");
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply scripts/validation.js:9]: From my local machine

File: contracts/vault/strategies/StrategyCompound.sol [hunk 10]
```diff
@@ -73,6 +73,7 @@ contract StrategyCompound is StrategyCore, IStrategyCompound {
         return (
             IporMath.division(
                 (shareToken.exchangeRateStored() * shareToken.balanceOf(address(this))),
+                //TODO: pogadać z rzonsą
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply contracts/vault/strategies/StrategyCompound.sol]: done

File: test/endToEnd/StanleyAaveUsdtTest.ts [hunk 11]
```diff
@@ -5,8 +5,10 @@ const usdtAbi = require("../../abis/usdtAbi.json");
 const comptrollerAbi = require("../../abis/comptroller.json");
 const aaveIncentiveContractAbi = require("../../abis/aaveIncentiveContract.json");
 
-const zero = BigNumber.from("0");
-const one = BigNumber.from("1000000000000000000");
+const ZERO = BigNumber.from("0");
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply test/endToEnd/StanleyAaveUsdtTest.ts:8]: done

File: contracts/vault/Stanley.sol [hunk 12]
```diff
@@ -286,24 +253,24 @@ abstract contract Stanley is
...
-            from = address(strategyAave);
-            uint256 shares = strategyAave.balanceOf();
+            from = strategyAave;
+            uint256 shares = IStrategy(strategyAave).balanceOf();
             require(shares > 0, IporErrors.VALUE_NOT_GREATER_THAN_ZERO);
-            strategyAave.withdraw(shares);
+            IStrategy(strategyAave).withdraw(shares);
         }
 
         uint256 amount = ERC20Upgradeable(_asset).balanceOf(address(this));
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply contracts/vault/Stanley.sol:276]: You right, fixed

[the remark this answers is not in the document]
Comment (mario-ipor) [reply contracts/vault/Stanley.sol:276]: I changed to IERC20Upgradeable and I used IERC20MetadataUpgradeable in places where decimals() method is required

**Pull request merged**